drag-and-drop/library/src/utils/runtime-flow-state.ts

```diff
@@ -188,6 +191,105 @@ export class RuntimeFlowState {
     asset.value = undefined;
   }
 
+  createDropTarget(
+    pluginName: string,
+    targetAssetType: ObjectType,
+    targetAsset: Asset,
+    propName = '',
+    parentAssetType = ''
+  ): DropTargetAssetType {
+    const dropTarget: DropTargetAssetType = {
+      id: uuidv4(),
+      __type: DragAndDropAssetType,
+      type: 'drop-target',
+      value: {
+        identifier: {
+          pluginName,
+          name: targetAssetType.name || '',
+          capability: parentAssetType.length === 0 ? 'Views' : 'Assets',
+        },
+        type: targetAssetType,
+        asset: targetAsset,
+      },
+    };
+
+    if (parentAssetType.length > 0 && propName.length > 0) {
+      dropTarget.context = {
+        propertyName: propName,
+        parent: {
+          pluginName,
+          name: parentAssetType,
+        },
+      };
+    }
+
+    this.assetMappings[dropTarget.id] = dropTarget;
+    return dropTarget;
+  }
+
+  addDropTarget(
+    assetWrapper: AssetWrapper,
+    xlrService: XLRService,
+    pluginName: string,
+    propName: string,
+    parentAssetType: string
+  ) {
+    const targetAsset = assetWrapper.asset;
+    const targetAssetType = xlrService.XLRSDK.getType(
+      targetAsset.type
+    ) as ObjectType;
+    /* eslint-disable-next-line no-param-reassign */
+    assetWrapper.asset = this.createDropTarget(
+      pluginName,
+      targetAssetType,
+      targetAsset,
+      propName,
+      parentAssetType
+    );
+    this.addDndStateToAsset(targetAsset, xlrService, pluginName);
+  }
+
+  addDndStateToAsset(asset: Asset, xlrService: XLRService, pluginName: string) {
+    const assetType = xlrService.XLRSDK.getType(asset.type) as ObjectType;
+    if (assetType === undefined || assetType.name === undefined) {
+      throw new Error(`cannot find asset type: ${asset.type}`);
+    }
+
+    Object.entries(asset).forEach(([key]) => {
+      if (key in assetType.properties) {
+        const { node } = assetType.properties[key];
+        if (node.type === 'ref' && node.ref.startsWith('AssetWrapper')) {
+          this.addDropTarget(
+            asset[key] as AssetWrapper,
+            xlrService,
+            pluginName,
+            key,
+            asset.type
+          );
+        } else if (
+          node.type === 'array' &&
+          node.elementType.type === 'ref' &&
+          node.elementType.ref.startsWith('AssetWrapper')
+        ) {
+          (asset[key] as Array<AssetWrapper>).forEach((value: AssetWrapper) => {
+            this.addDropTarget(value, xlrService, pluginName, key, asset.type);
+          });
+        }
+      }
+    });
+  }
+
+  importView(view: View, xlrService: XLRService, pluginName: string) {
+    const viewType = xlrService.XLRSDK.getType(view.type) as ObjectType;
```

Comment:
Do we need to fetch the XLR type for `view` here if it is also fetched in `addDndStateToAsset`?